sycl/source/detail/persistent_device_code_cache.cpp

```diff
@@ -557,6 +557,53 @@ void PersistentDeviceCodeCache::putCompiledKernelToDisc(
     updateCacheFileSizeAndTriggerEviction(getRootDir(), TotalSize);
 }
 
+void PersistentDeviceCodeCache::putDeviceCodeIRToDisc(
+    const std::string &Key, const std::vector<char> &IR) {
+
+  repopulateCacheSizeFile(getRootDir());
+
+  // Do not insert any new item if eviction is in progress.
+  // Since evictions are rare, we can afford to spin lock here.
+  const std::string EvictionInProgressFile =
+      getRootDir() + EvictionInProgressFileSuffix;
+  // Stall until the other process finishes eviction.
+  while (OSUtil::isPathPresent(EvictionInProgressFile))
+    continue;
+
+  // Total size of the item that we are writing to the cache.
+  size_t TotalSize = 0;
+
+  std::string DirName = getDeviceCodeIRPath(Key);
+
+  try {
+    OSUtil::makeDir(DirName.c_str());
+    std::string FileName = DirName + "/ir";
+    std::string FullFileName = FileName + ".bin";
```

Comment:
Is the file we write a valid LLVM IR Bitcode file? If so, would it make sense to use the file ending `.bc` here?

Reply:
Yes it is a `.bc` file, but a) I wasn't sure if we want to advertise that, and b) we'd need to extend to eviction logic to check for the third extension (so always one FS access more).